tools/gfx-unit-test/get-buffer-handle-test.cpp

```diff
@@ -0,0 +1,87 @@
+#include "tools/unit-test/slang-unit-test.h"
+
+#include "slang-gfx.h"
+#include "gfx-test-util.h"
+#include "tools/gfx-util/shader-cursor.h"
+#include "source/core/slang-basic.h"
+
+#ifdef _WIN32
+#include <d3d12.h>
+#endif
+
+using namespace gfx;
+
+namespace gfx_test
+{
+    void getBufferHandleTestImpl(IDevice* device, UnitTestContext* context)
+    {
+        // We need to create a transient heap in order to create a command buffer.
+        Slang::ComPtr<ITransientResourceHeap> transientHeap;
+        ITransientResourceHeap::Desc transientHeapDesc = {};
+        transientHeapDesc.constantBufferSize = 4096;
+        GFX_CHECK_CALL_ABORT(
+            device->createTransientResourceHeap(transientHeapDesc, transientHeap.writeRef()));
+
+        auto commandBuffer = transientHeap->createCommandBuffer();
+        ICommandBuffer::NativeHandle handle = 0;
+        GFX_CHECK_CALL_ABORT(commandBuffer->getNativeHandle(&handle));
+        if (device->getDeviceInfo().deviceType == gfx::DeviceType::Vulkan)
+        {
+            SLANG_CHECK(handle != 0);
+        }
+        else
+        {
+            auto d3d12Handle = (ID3D12GraphicsCommandList*)handle;
+            Slang::ComPtr<ID3D12GraphicsCommandList4> testHandle;
```

Comment:
The query for `ID3D12GraphicsCommandList4` will not succeed on systems without ray tracing support. Try to query for `ID3D12CommandList` instead.